lib/core/jbatch.c

```diff
@@ -273,20 +273,32 @@ j_batch_get_semantics(JBatch* batch)
 {
 	J_TRACE_FUNCTION(NULL);
 
-	g_return_val_if_fail(batch != NULL, NULL);
+	if (batch == J_SINGLE_OP)
+	{
+		return j_semantics_new(J_SEMANTICS_TEMPLATE_DEFAULT);
+	}
```

Comment:
We could also allocate a semantics singleton so callers do not have to free the semantics. I'm not sure whether it's worth the effort, though.